src/resolvers/funding.ts

```diff
@@ -196,44 +195,68 @@ export const resolvers: Resolvers = {
       }
     },
 
-    // add a new plan funding
-    addPlanFunding: async (_, { planId, projectFundingId }, context: MyContext): Promise<PlanFunding> => {
-      const reference = 'addPlanFunding resolver';
+    // add new plan fundings
+    addPlanFunding: async (_, { planId, projectFundingIds }, context: MyContext): Promise<Plan> => {
+      const reference = 'addPlanFunding mutator';
+
       try {
         if (isAuthorized(context.token)) {
           const plan = await Plan.findById(reference, context, planId);
           if (isNullOrUndefined(plan)) {
             throw NotFoundError();
           }
 
-          const projectFunding = await ProjectFunding.findById(
-            reference,
-            context,
-            projectFundingId
-          );
-          if (isNullOrUndefined(projectFunding)) {
-            throw NotFoundError();
-          }
-
           const project = await Project.findById(reference, context, plan.projectId);
           if (await hasPermissionOnProject(context, project)) {
-            const newFunding = new PlanFunding({ planId, projectFundingId });
 
-            if (newFunding && !newFunding.hasErrors()) {
-              // Version all of the plans (if any) and sync with the DMPHub
-              const planVersion = await updateVersion(context, plan, reference);
-              if (!planVersion || planVersion.hasErrors()) {
-                newFunding.addError("general", "Unable to version the plan");
+            // Fetch all of the current Funders associated with this Plan
+            const fundings = await PlanFunding.findByPlanId(reference, context, plan.id);
+            const currentPlanFundingids = fundings ? fundings.map((d) => d.projectFundingId) : [];
+            // Use the helper function to determine which funders to add
+            const {
```

Comment:
We also need to remove here, so add `idsToBeRemoved` here and then a section below to process them:
```
// Delete any fundings that were removed
for (const id of idsToBeRemoved) {
  const funding = await PlanFunding.findById(reference, context, id);
  if (funding) {
    const wasRemoved = funding.delete(context)
    if (!wasRemoved) {
      failedFundings.push(`Unable to remove projectFundingId: ${id}`);
    }
  }
}
```